### PR TITLE
[CALCITE-5607] Error when deserializing TIMESTAMP_DIFF in RelJson#toJson

### PR DESCRIPTION
**Pull request by olivrlee**
https://issues.apache.org/jira/browse/CALCITE-5607

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/plan/RelWriterTest.java [hunk 1]
```diff
@@ -1052,6 +1053,41 @@ void testAggregateWithAlias(SqlExplainFormat format) {
     assertThat(result, isLinux(expected));
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5607">[CALCITE-5607]</a>
+   *
+   * <p>Before the fix, RelJson.toRex would throw an ArrayIndexOutOfBounds error
+   * when deserialization.
+   * */
+  @Test void testDeserializeMinusDateOperator() {
```
Comment (tjbanghart) [inline core/src/test/java/org/apache/calcite/plan/RelWriterTest.java:1062]: The commit message and test case don't seem to align. This checks that `MINUS` can be deserialized but I don't see anything to suggest we've changed how the return `type` is being serialized.

File: core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java [hunk 2]
```diff
@@ -622,6 +622,7 @@ public Object toJson(RexNode node) {
         }
         map.put("operands", list);
         switch (node.getKind()) {
+        case MINUS:
```
Comment (tjbanghart) [inline core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java:625]: We should change the Jira issue to reflect that we are only changing the behavior of `MINUS`.

File: core/src/test/java/org/apache/calcite/plan/RelWriterTest.java (hunk 1 above)
Comment (olivrlee) [reply core/src/test/java/org/apache/calcite/plan/RelWriterTest.java:1062]: Yeah let me update the JIRA summary and the ticket title for this.
We're changing it to only affect MINUS

**Review by tjbanghart — APPROVED** [review]
LGTM Thanks!

**Pull request merged**